pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,520 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clickhouseclient
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go"
+	"github.com/gammazero/deque"
+	ipfixentities "github.com/vmware/go-ipfix/pkg/entities"
+	"k8s.io/klog/v2"
+)
+
+const (
+	maxQueueSize = 1 << 19 // 524288. ~500MB assuming 1KB per record
```

Comment:
this should be configurable? or do you think there is no value in being able to tune it through config?

Reply:
the queue size was designed as an internal param. given that queue size, record size and flow record rate combined defines the max buffer retention time window, and a large queue size alone won't result in large memory consumption in a calm cluster, it's probably not a good idea to expose it as a configurable param directly to end user. the current deque pkg will also align the number to nearest value power of 2.  I think it could be a hassle to explain all these and recommendations in the doc and probably users won't really care.

Reply:
I tend to agree with @wsquan171 - this does not seem a parameter for which user could do a lot of tuning.